### PR TITLE
docs(store): create common patterns page

### LINKED ISSUE #2236: Docs: add docs on performing tasks based on dispatched actions within an active component
There is a common use case of wanting to perform some task inside a component based on actions that have been dispatched outside the component, normally with Effects. There are some common patterns to achieve this: 

Use the Effects APIs to listen to dispatched actions:

```ts
import { Actions, ofType } from '@ngrx/effects';

@Component({ ... })
export class ActiveComponent {
  destroy$ = new Subject();

  constructor(private actions$: Actions) {}

  ngOnInit() {
    this.actions$.pipe(
      ofType(someSuccessAction),
      takeUntil(this.destroy$),
      tap(() => {
        this.doSomethingLocally();
      })
    ).subscribe();
  }

  ngOnDestroy() {
    this.destroy$.next();
  }

  doSomethingLocally() {
  }
}
```

Set some property in your state slice that gets updated, and listen for when that piece of state is updated.

```ts
import { Actions, ofType } from '@ngrx/effects';

@Component({ ... })
export class ActiveComponent {
  destroy$ = new Subject();
  done$ = this.store.select(selectDoneFromStateSlice);

  constructor(private store: Store<{}>) {}

  ngOnInit() {
    this.done$.pipe(
      filter(done => !!done),
      takeUntil(this.destroy$),
      tap(() => {
        this.doSomethingLocally();
      })
    ).subscribe();
  }

  ngOnDestroy() {
    this.destroy$.next();
  }

  doSomethingLocally() {
  }
}
```

We should document a recommended way to do this for developers.

### PR DESCRIPTION
**Pull request by JimMorrison723**
Add Performing tasks based on dispatched actions/state changes sections based on #2236

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #2236

The patterns mentiond in the issue could not be found in the docs.

## What is the new behavior?

A new common page is now available under recipes (could not come up with a short title to be a separated page :( ). Basically a 1-1 transfer from the issue description, so these patterns would be available for everyone in the docs.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by briannarenni** [comment]
I just noticed the newer version of eslint was added to my branch's package.json and yarn.lock (`@ngrx/eslint-plugin@^15.1.0`). I can remove it if needed.

**Review by timdeschryver — COMMENTED** [review]
Hi, thanks for creating this PR.

I do think that it needs some more "body" to explain the use-case/example.
Do you want to add that here?

Also, make sure that the examples have all the imports, otherwise it can be hard to follow where the things are coming from. For example, the `someSuccessAction` action, or operators from RxJS.

Because it's an old issue, I think it would also be better to add a note that this easier with `@ngrx/component` and that that might be better to use for these use cases.

**Comment by brandonroberts** [comment]
I agree with Tim. This guide could be sections in other places. Closing for now

**Comment by JimMorrison723** [comment]
Ah sorry, i somehow missed the notifications. I'll get back to it hopefully this week and update the pr
